File: Guigna/Guigna/GPackage.m
#import "GPackage.h"
#import "GSystem.h"

@implementation GPackage

- (instancetype)initWithName:(NSString *)name
           version:(NSString *)version
            system:(GSystem *)system
            status:(GStatus)status {
    self = [super initWithName:name
            version:version
            source:(GSource *)system
            status:status];
    self.system = system;
    return self;
}

- (NSString *)key {
    return [self.system keyForPackage:self];  
}

- (NSArray *)dependenciesList {
    return [self.system dependenciesList:self];
}

- (NSString *)installCmd {
    return [self.system installCmd:self];
}


- (NSString *)uninstallCmd {
    return [self.system uninstallCmd:self];
}


- (NSString *)deactivateCmd {
    return [self.system deactivateCmd:self];
}


- (NSString *)upgradeCmd {
    return [self.system upgradeCmd:self];
}


- (NSString *)fetchCmd {
    return [self.system fetchCmd:self];
}


- (NSString *)cleanCmd {
    return [self.system cleanCmd:self];
}


@end

File: Guigna-Swift/Guigna/GPackage.swift
import Foundation

class GPackage: GItem {
    
    var options: String!
    var markedOptions: String!
    var repo: String?
    
    init(name: String, version: String, system: GSystem, status: GStatus) {
        super.init(name: name, version: version, source: system, status: status)
        self.system = system
    }
    
    var key: String  {
        get {
            return self.system.key(package: self)
        }
    }
    
    var installCmd: String {
        get {
            return self.system.installCmd(self)
        }
    }
    
    var uninstallCmd: String {
        get {
            return self.system.uninstallCmd(self)
        }
    }
    
    var deactivateCmd: String {
        get {
            return self.system.deactivateCmd(self)
        }
    }
    
    var upgradeCmd: String {
        get {
            return self.system.upgradeCmd(self)
        }
    }
    
    var fetchCmd: String {
        get {
            return self.system.fetchCmd(self)
        }
    }
    
    var cleanCmd: String {
        get {
            return self.system.cleanCmd(self)
        }
    }
}
